src/core_ocean/analysis_members/mpas_ocn_mixed_layer_heat_budget.F

```diff
@@ -0,0 +1,417 @@
+         call mpas_pool_get_array(diagnosticsPool, 'activeTracerNonLocalTendency', &
+                                  activeTracerNonLocalTendency)
+         call mpas_pool_get_array(diagnosticsPool, 'activeTracerVerticalAdvectionTendency', &
+                                  activeTracerVerticalAdvectionTendency)
+         call mpas_pool_get_array(diagnosticsPool, 'activeTracerHorizontalAdvectionTendency', &
+                                  activeTracerHorizontalAdvectionTendency)
+         call mpas_pool_get_array(diagnosticsPool, 'activeTracerVertMixTendency', &
+                                  activeTracerVertMixTendency)
+         call mpas_pool_get_array(diagnosticsPool, 'temperatureShortWaveTendency', &
+                                  temperatureShortWaveTendency)
+         call mpas_pool_get_array(diagnosticsPool, 'BruntVaisalaFreqTop', BruntVaisalaFreqTop)
+
+         call mpas_pool_get_array(statePool, 'layerThickness', layerThickness)
+
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerForcingMLTend', &
+                                  activeTracerForcingMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerNonLocalMLTend', &
+                                  activeTracerNonLocalMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerHorMixMLTend', &
+                                  activeTracerHorMixMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerVertMixMLTend', &
+                                  activeTracerVertMixMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerHorAdvectionMLTend', &
+                                  activeTracerHorAdvectionMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerVertAdvectionMLTend', &
+                                  activeTracerVertAdvectionMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracersML', activeTracersML)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'bruntVaisalaFreqML', bruntVaisalaFreqML)
+         call mpas_pool_get_array(mixedLayerDepthsAMPool, 'dThreshMLD', dThreshMLD)
+
+         call mpas_pool_get_array(tracersPool, 'activeTracers', tracers, 1)
+
+         num_tracers = size(activeTracerVertMixTendency, dim=1)
+
+         do iCell = 1, nCellsSolve
+            activeTracerHorAdvectionMLTend(:, iCell) = 0.0_RKIND
+            activeTracerNonLocalMLTend(:, iCell) = 0.0_RKIND
+            activeTracerHorMixMLTend(:, iCell) = 0.0_RKIND
+            activeTracerVertMixMLTend(:, iCell) = 0.0_RKIND
+            activeTracerVertAdvectionMLTend(:, iCell) = 0.0_RKIND
+            activeTracerForcingMLTend(:, iCell) = 0.0_RKIND
+            activeTracersML(:, iCell) = 0.0_RKIND
+            k = 1
+            depth = layerThickness(k, iCell)
+            do while (depth <= dThreshMLD(iCell) .and. k < maxLevelCell(iCell))
```

Comment:
quick question: can MLD ever be smaller than layerThickness(1)? or is it set to 0 in that case?

Reply:
the mixed layer will always be above 10m + the next layer thickness so this shouldn't be a problem, but is probably wise to fix just in case.

Reply:
actually looking at the code again, I think it is okay.  If layer dThreshMLD < layerTHickness(k,ICell) we will just pick up the partial integral after this do loop

Reply:
good point.